public/app/features/admin/UserListPage.tsx

```diff
@@ -10,31 +12,84 @@ import { AccessControlAction } from '../../types';
 import { UsersListPageContent } from '../users/UsersListPage';
 
 import { UserListAdminPageContent } from './UserListAdminPage';
+import { UserListPublicDashboardPage } from './UserListPublicDashboardPage/UserListPublicDashboardPage';
+
+enum TabView {
+  ADMIN = 'admin',
+  ORG = 'org',
+  PUBLIC_DASHBOARDS = 'public-dashboards',
+}
+
+const selectors = e2eSelectors.pages.UserListPage;
+
+const PublicDashboardsTab = ({ view, setView }: { view: TabView | null; setView: (v: TabView | null) => void }) => (
+  <Tab
+    label="Public dashboard users"
+    active={view === TabView.PUBLIC_DASHBOARDS}
+    onChangeTab={() => setView(TabView.PUBLIC_DASHBOARDS)}
+    data-testid={selectors.tabs.publicDashboardsUsers}
+  />
+);
+
+const TAB_PAGE_MAP: Record<TabView, React.ReactElement> = {
+  [TabView.ADMIN]: <UserListAdminPageContent />,
+  [TabView.ORG]: <UsersListPageContent />,
+  [TabView.PUBLIC_DASHBOARDS]: <UserListPublicDashboardPage />,
+};
 
 export default function UserListPage() {
+  const styles = useStyles2(getStyles);
+
   const hasAccessToAdminUsers = contextSrv.hasAccess(AccessControlAction.UsersRead, contextSrv.isGrafanaAdmin);
   const hasAccessToOrgUsers = contextSrv.hasPermission(AccessControlAction.OrgUsersRead);
-  const styles = useStyles2(getStyles);
+  const hasEmailSharingEnabled =
+    Boolean(config.featureToggles.publicDashboards) &&
+    Boolean(config.featureToggles.publicDashboardsEmailSharing) &&
+    featureEnabled('publicDashboardsEmailSharing');
+
   const [view, setView] = useState(() => {
     if (hasAccessToAdminUsers) {
-      return 'admin';
+      return TabView.ADMIN;
     } else if (hasAccessToOrgUsers) {
-      return 'org';
+      return TabView.ORG;
     }
     return null;
   });
 
-  const showToggle = hasAccessToOrgUsers && hasAccessToAdminUsers;
+  const showAdminAndOrgTabs = hasAccessToOrgUsers && hasAccessToAdminUsers;
```

Comment:
Do you need access to both things to see both tags?

I mean:
* Can you see the Orgs tab if you have just access to org users? 

This seems weird to me since both are separate permissions but seem to be independent. But maybe I don't know the business logic behind it.

Maybe to simplify this, we can move all the tabs into an array. If the length is > 0, then I show tabs, otherwise anything else.

Something like:
```
const tabs= [];

if(hasAccessToAdminUsers) {
   tabs.push (<Tab ... />)
}

// ...

{ tabs.length > 0 ? <TabsBar className={styles.tabsMargin}>
  {tabs}
</TabsBar> : <Whatever/>
```

Reply:
This is logic already present in this component. It shows a distinction between on Prem and on Cloud

Reply:
For example, in your own cloud instance you are not the admin of Grafana. In that case and because just one tab would be present (the Organization users one, which besides doesn't make sense because there's just one org in cloud), the code doesn't show any tab. I think that's the reason of this flag.

Reply:
In your example with a tab array, we would need an if and else sentence and in that case I prefer this solution

Reply:
Ok, cool! Let's leave it like this then.